frame/example-offchain-worker/src/lib.rs

```diff
@@ -268,71 +328,155 @@ impl<T: Trait> Module<T> {
+		})?;
 
-		// Create a str slice from the body.
-		let body_str = sp_std::str::from_utf8(&body).map_err(|_| {
-			debug::warn!("No UTF8 body");
+		let fetch_partition_list_str = sp_std::str::from_utf8(&fetch_partition_list_bytes).map_err(|_| {
+			debug::warn!("fetch_partition_list_str: No UTF8 body");
 			http::Error::Unknown
 		})?;
+		// debug::info!("fetch_partition_list_str: {}", fetch_partition_list_str);
 
-		debug::info!("body_str: {}", body_str);
 		// Parse the string of data into serde_json::Value.
-		let node_info: Vec<NodeInfo> = serde_json::from_str(&body_str).map_err(|_| {
-			debug::warn!("No UTF8 body");
+		let partition_info: Vec<PartitionInfo> = serde_json::from_str(&fetch_partition_list_str).map_err(|_| {
+			debug::warn!("Parse body to Vec<PartitionInfo> error");
 			http::Error::Unknown
 		})?;
+		debug::info!("Partition_info length: {:#?}", partition_info.len());
+
+		let mut agreated_result: Vec<MetricInfo> = Vec::new();
+		for one_partition in partition_info.iter() {
+			let id_from_partition = sp_std::str::from_utf8(&one_partition.nodeId).unwrap();
 
-		// debug::info!("Nodes info: {:#?}", node_info);
+			let mut node_info_iter = node_info.iter(); // id is Vec<u8>
+			let equal_index = node_info_iter.position(|node| id_from_partition.eq(sp_std::str::from_utf8(&node.id).unwrap()));
+			// debug::info!("id_from_partition: {}", id_from_partition);
+			debug::info!("equal_index: {:?}", &equal_index);
 
-		for one_node in node_info.iter() {
-			debug::info!("Nodes info: {:?}", one_node);
-			let resp_bytes = Self::fetch_from_node(one_node).map_err(|e| {
-				debug::error!("fetch_from_node error: {:?}", e);
+			if equal_index.is_none() {
+				debug::info!("Can not found in nodeId {:?} in the topology", sp_std::str::from_utf8(&one_partition.nodeId).unwrap());
+				continue;
+			}
+
+			let metrics_url = sp_std::str::from_utf8(&node_info[equal_index.unwrap()].httpAddr).map_err(|_| {
+				debug::warn!("httpAddr");
+				http::Error::Unknown
+			})?;
+
+			let app_pubkey_str = sp_std::str::from_utf8(&one_partition.appPubKey).map_err(|_| {
+				debug::warn!("appPubKey error");
 				http::Error::Unknown
 			})?;
+			let metrics_url_with_key = format!("{}{}{}", metrics_url, HTTP_METRICS, app_pubkey_str);
+			// debug::info!("metrics_url_with_key: {}", metrics_url_with_key);
 
-			let resp_str = sp_std::str::from_utf8(&resp_bytes).map_err(|_| {
-				debug::warn!("No UTF8 body");
+			let partition_id_str = sp_std::str::from_utf8(&one_partition.id).map_err(|_| {
+				debug::warn!("partition id error");
 				http::Error::Unknown
 			})?;
 
-			// Print out our fetched JSON string
-			debug::info!("Node data: {}", resp_str);
+			let metrics_url_with_partition = format!("{}{}{}", metrics_url_with_key, METRICS_PARAM_PARTITIONID, partition_id_str);
+			// debug::info!("metrics_url_with_partition: {}", metrics_url_with_partition);
 
-			// Send signed Tx, TODO: send to call SC method.
+			let last_time_stamp: u64 = 1619690465923;
```

Comment:
For this ticket, just use `now - 24hours`

Reply:
Fixed. @naure